src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/QueryTag.cs

```diff
@@ -90,6 +90,11 @@ public QueryTag(WorkitemQueryTagStoreEntry entry)
         /// </summary>
         public WorkitemQueryTagStoreEntry WorkitemQueryTagStoreEntry { get; }
 
+        /// <summary>
+        /// Gets whether this is workitem query tag or not.
+        /// </summary>
+        public bool IsWorkitemQueryTag => WorkitemQueryTagStoreEntry != null;
```

Comment:
We need to clean all this up in V2. QueryTag should come out of ExtendedQueryTag and not include Any workItem or ExtendedQueryTag info. 

See in the V2 if you are already defending against the change of concept

Reply:
Totally agree, we dont want to delay this. Since we have done most in iteration 1, we can start the refactoring righaway